lib/ansible/modules/cloud/amazon/ecs_service.py

```diff
@@ -466,7 +481,8 @@ def main():
             security_groups=dict(type='list'),
             assign_public_ip=dict(type='bool'),
         )),
-        launch_type=dict(required=False, choices=['EC2', 'FARGATE'])
+        launch_type=dict(required=False, choices=['EC2', 'FARGATE']),
+        health_check_grace_period_seconds=dict(required=False, type='int', default=30)
```

Comment:
Isn't healthcheck grace period 0 seconds by default?  Why would it default to 30 here?

Reply:
Couldn't find any docs on what values are allowed. However I assume I had a reason setting it to 30, probably if setting it there is a minimum. It's long ago I actually implented this, but I can recheck if you want.

Reply:
From https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.create_service

healthCheckGracePeriodSeconds (integer) -- The period of time, in seconds, that the Amazon ECS service scheduler should ignore unhealthy Elastic Load Balancing target health checks after a task has first started. This is only valid if your service is configured to use a load balancer. If your service's tasks take a while to start and respond to Elastic Load Balancing health checks, you can specify a health check grace period of up to 7,200 seconds during which the ECS service scheduler ignores health check status. This grace period can prevent the ECS service scheduler from marking tasks as unhealthy and stopping them before they have time to come up.

I've also checked some of my services that were created before health check grace period was added and the return for it is 0.  Having 30 seconds be the default wouldn't be a "problem" for me, but it would be a change from what I think is current behavior and not what I would expect.

Reply:
Yes, I saws those docs already unfortunaltely they didn't say anything about the minimum value. But if you say it's 0 I will simply change that.

Reply:
However looking at the code again it's actually an optional value, hence if you don't provide the parameter `health_check_grace_period_seconds` it won't actually be set (no matter what the default value is). I guess I just followed the pattern of the few lines above to always set a sane default if the param is `required=False`.

Reply:
Just to clarify, I ran through a quick test omitting health check grace period with default at 30 and it looks like it gets set even when I omit in yaml.

But I see you dropped it, so I think this looks good.  I'll give it another look over shortly and put a full review in if I can.

Thanks.